Manager/Manager.java

```diff
@@ -38,273 +39,278 @@
         Filter tagFilter = new Filter("tag:Type", Arrays.asList("Worker"));
         Filter stateFilter = new Filter("instance-state-name", Arrays.asList("pending", "running"));
 
         DescribeInstancesResult result = ec2.describeInstances((new DescribeInstancesRequest())
-        		.withFilters(tagFilter, stateFilter));
+                .withFilters(tagFilter, stateFilter));
         logInfo("checking for workers");
-        
+
         int numOfCurrentWorkers = 0;
         if(result != null && !result.getReservations().isEmpty())
-        	numOfCurrentWorkers = getNumOfCurrentWorkers(result);
-        
+            numOfCurrentWorkers = getNumOfCurrentWorkers(result);
+
         int numOfRequiredWorkers = getNumOfRequiredWorkers();
         logInfo("currentWorkers = " + numOfCurrentWorkers + ". neededWorkers = " + numOfRequiredWorkers);
         BalanceWorkers(numOfRequiredWorkers - numOfCurrentWorkers);
         logInfo("Done round of management");
-	}
-	private static void BalanceWorkers(int balance) {
-		logInfo("Balancing workers. Got - " + balance);
-		if(balance < 0)
-			fireWorkers((-1) * balance);
-		else if(balance > 0)
-			recruitWorkers(balance);
-	}
-
-	private static void recruitWorkers(int numOfWorkers) {
-		logInfo("Creating new " + numOfWorkers + " Workers");
-		int recruited = WorkerInstanceData.getWorkers(numOfWorkers, ec2);
-		if(recruited != numOfWorkers){
-			logInfo("Couldn't create required amount of workers. Created " + recruited + " instead of " + numOfWorkers);
-			manage();
-		}
-	}
-
-	private static void fireWorkers(int numOfWorkers) {
-		logInfo("Firing " + numOfWorkers + " Workers");
-		for(int i = 0; i < numOfWorkers; i++)
-			sqs.sendMessage(new SendMessageRequest().withQueueUrl(managerWorkersQueue).withMessageBody("You Are Fired!").withMessageAttributes(fireWorkerMsg));
-	}
-
-	private static int getNumOfCurrentWorkers(DescribeInstancesResult result) {
-		int numOfCurrentWorkers = 0;
-		for (int i = 0; i < result.getReservations().size(); i++) {
-        	numOfCurrentWorkers += result.getReservations().get(i).getInstances().size();
+    }
+    private static void BalanceWorkers(int balance) {
+        logInfo("Balancing workers. Got - " + balance);
+        if(balance < 0)
+            fireWorkers((-1) * balance);
+        else if(balance > 0)
+            recruitWorkers(balance);
+    }
+
+    private static void recruitWorkers(int numOfWorkers) {
+        logInfo("Creating new " + numOfWorkers + " Workers");
+        int recruited = WorkerInstanceData.getWorkers(numOfWorkers, ec2);
+        if(recruited != numOfWorkers){
+            logInfo("Couldn't create required amount of workers. Created " + recruited + " instead of " + numOfWorkers);
+            manage();
         }
-		return numOfCurrentWorkers;
-	}
-
-	private static int getNumOfRequiredWorkers(){
-		int numOfRequired = 0;
-		for(String uuid : clientsUUIDToURLLeft.keySet()){//TODO: CHECK WHY requiredWorkers.. is null.. 
```

Comment:
shouldn't we also use the worker ratio? or is required worker per task already doing that?